pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshiftVersion"`
```

Comment:
@avishayt notes that this field would contain something like "4.6" here: https://github.com/openshift/enhancements/pull/574/files#r553131459 and would map to a key in a configmap such as:

```json
{
        "4.6": {
                "display_name": "4.6.8",
                "release_image": "quay.io/openshift-release-dev/ocp-release:4.6.8-x86_64",
                "rhcos_image": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.8/rhcos-4.6.8-x86_64-live.x86_64.iso",
                "support_level": "production"
        },
        "4.7": {
                "display_name": "4.7-pre-release",
                "release_image": "quay.io/openshift-release-dev/ocp-release@sha256:2419f9cd3ea9bd114764855653012e305ade2527210d332bfdd6dbdae538bd66",
                "rhcos_image": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.8/rhcos-4.6.8-x86_64-live.x86_64.iso",
                "support_level": "beta"
        }
}
```

Hive's closest concept to this is the [ClusterImageSet](https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1/clusterimageset_types.go#L27), which currently has only one field which is release image. ClusterDeployments link to one via 

Can rhcos_image be determined by inspecting the release_image contents? If so, this is a very good parallel to what Hive does today, before launching an install, we extract the installer container image from the release image, so we can run that in our pod that performs the openshift-install. We could use the same path to also extract the rhcosImage, and set it on ClusterDeployment. Status similar to [this](https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1/clusterdeployment_types.go#L223).

I've always wanted to instead store this only once per ClusterImageSet in it's status, you set the release image, and the controllers populate the status for the installer image, cli image, and now rhcos image, but this hasn't happened yet. Would be a nice optimization though and maybe we can start doing it this way here.

Reply:
Note that this configmap is used by the service, not the user.  The user asks for 4.6.  The service uses the information in the configmap to know how to provide that.

Reply:
This is how ClusterImageSet works as well, they are configured centrally but management for the Hive service, people creating clusters can then reference them.

Reply:
We agreed on call that ClusterImageSet is effectively authoritative. There are a lot of open questions on how this flows to assisted, whether we can override and pass each setting to assisted installer, or we automatically generate the configmap for them, or if there should be some tighter integration between assisted controllers working directly with ClusterImageSet directly.

However for purposes of API, we will rely on ClusterDeployment.Spec.ClusterImageSetRef and work our the rest of the details beneath the covers over time.